keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -0,0 +1,169 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.ArtifactSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactSummaryInEnvironment
+import com.netflix.spinnaker.keel.core.api.ArtifactVersionSummary
+import com.netflix.spinnaker.keel.core.api.ArtifactVersions
+import com.netflix.spinnaker.keel.core.api.EnvironmentSummary
+import com.netflix.spinnaker.keel.pause.ActuationPauser
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.ResourceStatus
+import com.netflix.spinnaker.keel.persistence.ResourceSummary
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * Service object that offers high-level APIs for application-related operations.
+ */
+@Component
+class ApplicationService(
+  private val pauser: ActuationPauser,
+  private val repository: KeelRepository,
+  private val artifactRepository: ArtifactRepository
```

Comment:
isn't the artifact repository in the keel repository?

Reply:
What's the difference between `KeelRepository` and `CombinedRepository`? Is the former the new name for the latter or are they different things?

Reply:
`KeelRepository` is the interface, `CombinedRepository` the (only) implementation.

@emjburns It is, but I added new methods which are not used anywhere else but here so it seemed unnecessary to expose them in `KeelRepository`. I can add them if you think it makes sense.

Reply:
P.S. One of the reasons I added this new `ApplicationService` class is that I feel this kind of coordination among multiple repositories would be best encapsulated in service objects like these. Thoughts?